apps/fz_http/lib/fz_http_web/controllers/auth_controller.ex

```diff
@@ -131,16 +144,23 @@ defmodule FzHttpWeb.AuthController do
 
   def redirect_oidc_auth_uri(conn, %{"provider" => provider}) do
     openid_connect = Application.fetch_env!(:fz_http, :openid_connect)
+    state = FzCrypto.rand_string()
 
     params = %{
-      state: FzCrypto.rand_string(),
+      state: state,
       # needed for google
       access_type: "offline"
     }
 
-    uri = openid_connect.authorization_uri(String.to_atom(provider), params)
+    uri = openid_connect.authorization_uri(String.to_existing_atom(provider), params)
 
     conn
+    |> put_resp_cookie(@oidc_state_key, state,
+      max_age: @oidc_state_valid_duration,
+      sign: true,
+      same_site: "Lax",
+      secure: true
```

Comment:
For this field, could you pull it from `Application.fetch_env!(:fz_http, :cookie_secure)`? @conectado was working on deployment docs using different reverse proxies and determined it's helpful to be able to run Phoenix over HTTP, so we're trying to make cookie attributes configurable.

```suggestion
      secure: Application.fetch_env!(:fz_http, :cookie_secure)
```

Reply:
Also, do you think it's a good idea to add to add [`signing_salt` and/or `encryption_salt`](https://hexdocs.pm/plug/Plug.Session.COOKIE.html#module-options) here? We use encrypted cookies in `apps/fz_http/lib/fz_http_web/session.ex` but not sure if that's overkill for this application.

Just thinking through some MITM scenarios

Reply:
[Here](https://github.com/firezone/firezone/pull/792/files#diff-2ca8cfd5ee77a5fc6cab45bca86fe75026eafc6113be1303994bdd3e019bb2c1R526) I added the config parameter in the branch I'm working for the new proxies. Made it a dev-only parameter like we discussed yesterday so that for non-dev we always require terminating SSL on the proxy.

Reply:
When https is on, signing is enough.

`put_resp_cookie` uses the cookie name to derive a signing salt.